core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -626,11 +624,10 @@ private[log] class Cleaner(val id: Int,
    * @param segment The segment to index
    * @param map The map in which to store the key=>offset mapping
    *
-   * @return The final offset covered by the map or -1 if the map is full
+   * @return If the map was filled whilst loading from this segment
    */
-  private def buildOffsetMapForSegment(topicAndPartition: TopicAndPartition, segment: LogSegment, map: OffsetMap): Long = {
+  private def buildOffsetMapForSegment(topicAndPartition: TopicAndPartition, segment: LogSegment, map: OffsetMap, start: Long): Boolean = {
```

Comment:
Since we won't always be starting from offset 0 now, would it be better to use the offset index to find the starting position close to offset "start" and start iterating from there? This could save the scanning if start is not 0.